src/GafferUI/ViewportGadget.cpp

```diff
@@ -770,16 +773,14 @@ ViewportGadget::ViewportGadget( GadgetPtr primaryChild )
 
 ViewportGadget::~ViewportGadget()
 {
-	if( m_framebufferSize != Imath::V2i( -1 ) )
+	// We should technically ensure that the right GL context is current when
+	// making these calls, but this seems to work without, because all our GL
+	// contexts are sharing the same resources.
+	if( m_framebuffer )
 	{
-		// We should technically ensure that the GL context is current when making these calls, but it
-		// seems to be working to just make them in the destructor.
-		// John says "I think we're just generally lucky because all our GL contexts are sharing
-		// the same resources, so it doesn't matter which is current"
-
-		glDeleteTextures(1, &m_framebufferTexture);
-		glDeleteRenderbuffers(1, &m_rbo);
-		glDeleteFramebuffers(1, &m_fbo);
+		glDeleteFramebuffers( 1, &m_framebuffer );
+		glDeleteTextures( 1, &m_framebufferTexture );
+		glDeleteRenderbuffers( 1, &m_depthBuffer );
```

Comment:
I stared at this for a while ... changing the order could probably have an effect, but it appears the old order was wrong, and this is correct.